{"name": "用户名错误", "status": "broken", "statusDetails": {"message": "TypeError: a bytes-like object is required, not 'dict'", "trace": "self = <autotest03.testcases.test_my_interface01.test_login.TestLogin object at 0x044D6250>\ncase = <comms.excel_utils.CaseData object at 0x04078F40>\n\n    @allure.severity(\"critical\")  # 设置用例优先级\n    @pytest.mark.parametrize(\"case\", cases)\n    def test_login(self, case):  # 所有的测试方法必须以test开头\n        allure.dynamic.title(case.case_title)  # 动态获取用例名称,allure报告显示用例名称\n        allure.attach(body=case.url, name='接口路径')\n        allure.attach(body=case.case_data, name='请求参数')\n        # 正确流程\n        if case.case_id == 1:\n            username = eval(case.case_data)[\"username\"]\n            password = eval(case.case_data)[\"password\"]\n            db = DBUtils()\n            db.cud('delete from tb_user where name = %s', (username,))\n            db.cud('insert into tb_user(name,passwd,email,phone) values(%s,%s,%s,%s)',\n                   (username, password, 'test@163.com', '18877446699'))\n            db.close()\n    \n        response = requests.post(url=case.url,\n                                 data=eval(case.case_data))  # 识别()中的python表达式，把字符串转成字典\n        res_body = response.json()  # res_body就是接口响应的实际结果\n>       allure.attach(body=res_body, name='响应结果')\n\ntest_login.py:44: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n..\\..\\..\\..\\新建文件夹\\lib\\site-packages\\allure_commons\\_allure.py:179: in __call__\n    plugin_manager.hook.attach_data(body=body, name=name, attachment_type=attachment_type, extension=extension)\n..\\..\\..\\..\\新建文件夹\\lib\\site-packages\\pluggy\\_hooks.py:265: in __call__\n    return self._hookexec(self.name, self.get_hookimpls(), kwargs, firstresult)\n..\\..\\..\\..\\新建文件夹\\lib\\site-packages\\pluggy\\_manager.py:80: in _hookexec\n    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)\n..\\..\\..\\..\\新建文件夹\\lib\\site-packages\\allure_pytest\\listener.py:222: in attach_data\n    self.allure_logger.attach_data(uuid4(), body, name=name, attachment_type=attachment_type, extension=extension)\n..\\..\\..\\..\\新建文件夹\\lib\\site-packages\\allure_commons\\reporter.py:119: in attach_data\n    plugin_manager.hook.report_attached_data(body=body, file_name=file_name)\n..\\..\\..\\..\\新建文件夹\\lib\\site-packages\\pluggy\\_hooks.py:265: in __call__\n    return self._hookexec(self.name, self.get_hookimpls(), kwargs, firstresult)\n..\\..\\..\\..\\新建文件夹\\lib\\site-packages\\pluggy\\_manager.py:80: in _hookexec\n    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n\nself = <allure_commons.logger.AllureFileLogger object at 0x0351DF70>\nbody = {'code': 1003, 'msg': '用户名或密码错误'}\nfile_name = '69255a79-9d2b-4ac5-9fac-1e2d5d1e5774-attachment.attach'\n\n    @hookimpl\n    def report_attached_data(self, body, file_name):\n        destination = os.path.join(self._report_dir, file_name)\n        with open(destination, 'wb') as attached_file:\n            if isinstance(body, text_type):\n                attached_file.write(body.encode('utf-8'))\n            else:\n>               attached_file.write(body)\nE               TypeError: a bytes-like object is required, not 'dict'\n\n..\\..\\..\\..\\新建文件夹\\lib\\site-packages\\allure_commons\\logger.py:62: TypeError"}, "attachments": [{"name": "接口路径", "source": "e7529652-a7cc-41e1-9616-75b59520dfb8-attachment.attach"}, {"name": "请求参数", "source": "05398feb-1bc9-488a-87e9-569ff1300a27-attachment.attach"}, {"name": "响应结果", "source": "69255a79-9d2b-4ac5-9fac-1e2d5d1e5774-attachment.attach"}], "parameters": [{"name": "case", "value": "<comms.excel_utils.CaseData object at 0x04078F40>"}], "start": 1646039328204, "stop": 1646039328210, "uuid": "ca415d40-4376-4c0c-9e06-c3b5ee9e5185", "historyId": "9e59e38f75c692b8b2657fbe9ccbbea2", "testCaseId": "b62a022e3184708d2d40012e446c9b1d", "fullName": "test_login.TestLogin#test_login", "labels": [{"name": "feature", "value": "登录接口测试"}, {"name": "severity", "value": "critical"}, {"name": "suite", "value": "test_login"}, {"name": "subSuite", "value": "TestLogin"}, {"name": "host", "value": "DESKTOP-V37VTV5"}, {"name": "thread", "value": "65852-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "test_login"}]}